relayer/chains/cosmos/cosmos_chain_processor.go

```diff
@@ -190,14 +207,40 @@ func (ccp *CosmosChainProcessor) queryCycle(ctx context.Context, persistence *qu
 
 			ccp.log.Debug("Parsed IBC messages", zap.Any("messages", messages))
 
-			// TODO pass messages to handlers
+			for _, m := range messages {
+				if handler, ok := messageHandlers[m.messageType]; ok {
+					// call message handler for this ibc message type. can do things like cache things on the chain processor or retain ibc messages that should be sent to the PathProcessors.
+					handlerMadeChangesForPathProcessor := handler(ccp, MsgHandlerParams{
+						messageInfo:   m.messageInfo,
+						foundMessages: foundMessages,
+					})
+					if handlerMadeChangesForPathProcessor {
+						haveChangesForPathProcessor = true
+					}
+				}
+			}
```

Comment:
If we do away with the `messagesHandlers` map as I suggest in a different comment, this would be more like

```go
for _, m := range messages {
  changed := ccp.handleMessage(m, ...)
  ppChanged = ppChanged || changed
}
```